### PR TITLE
Feature/aut 1647 add xinclude loading on sleep

### PR DESCRIPTION
**Pull request by Karol-Stelmaczonek**
Ticket: https://oat-sa.atlassian.net/browse/AUT-1647
## What's Changed

- Add loading of xi:include elements nested in pages

## Dependencies PRs

- https://github.com/oat-sa/extension-tao-test-preview-ui-loader/pull/166
- https://github.com/oat-sa/extension-tao-itemqti/pull/2466

## How to test
- Create new item
-  Add Text Reader Interaction
- Add shared stimulus to pages 
- Shared stimulus is rendered in authoring 

https://github.com/oat-sa/extension-tao-itemqti/assets/118974926/733898a3-79f3-43f8-859e-4ee7afb1f96a

### CODE REVIEW

**Review by jsconan — CHANGES_REQUESTED** [review]
The code looks ok. In the end, we should consider move the `xincludeloader` to a shared place so that we can reuse it.

Please add the PCI version bump and a migration script. Also, please compile the PCI and add the bundles.

**Review by jsconan — CHANGES_REQUESTED** [review]
Wait, in the authoring, there is an error in the console if the PCI does not come with a shared stimulus...
image

**Comment by viktar-dzmitryieu-tao** [comment]
It works ok, but:
- user can still edit the stimulus content in item-authoring, but the changes will not be applied which is correct. Probably we should lock ability to edit stimulus here as it can confuse the user.
- if we export the item with stimulus - `qti.xml` contains `xi:include href=\"taomedia://mediamanager..` in pages which means it couldn't be loaded as is on the other environment, I think we should consider the need of BE work on export functionality.

File: views/js/pciCreator/ims/textReaderInteraction/creator/js/xincludeLoader.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    function processContentItem(contentItem, baseUrl) {
+        const tempDiv = document.createElement('div');
+        tempDiv.innerHTML = contentItem;
+        const xiIncludeElements = tempDiv.querySelectorAll('xi\\:include');
+
+        const xiIncludePromises = Array.from(xiIncludeElements).map(xiElement => {
+            return loadXIncludeElement($(xiElement), baseUrl).then(newContent => {
+                xiElement.replaceWith(newContent);
+            });
```
[the remark this answers is not in the document]
Comment (Karol-Stelmaczonek) [reply views/js/pciCreator/ims/textReaderInteraction/creator/js/xincludeLoader.js]: Removed the jQuery

**Review by jsconan — COMMENTED** [review]
Good job, concise and neat!

**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code respects code style rules
- [x] New code respects best practices
- [ ] ~~New code is not subject to concurrency issues (if applicable)~~
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic

**Review by jsconan — CHANGES_REQUESTED** [review]
Wait, there is an issue as spotted by @viktar-dzmitryieu-tao.
Actually, the feature does:
- the shared stimulus is inlined inside the PCI…
- user can modify the content and save it
- the PCI can be exported and imported in another platform: the shared stimulus being inlined, the content is present but the shared stimulus is not defined as such
- any change made later to the shared stimulus is not reflected… because it is inlined

Finally, there is a problem because the connection with the shared stimulus is lost at some point

**Comment by Karol-Stelmaczonek** [comment]
> It works ok, but:
> 
> * user can still edit the stimulus content in item-authoring, but the changes will not be applied which is correct. Probably we should lock ability to edit stimulus here as it can confuse the user.
> * if we export the item with stimulus - `qti.xml` contains `xi:include href=\"taomedia://mediamanager..` in pages which means it couldn't be loaded as is on the other environment, I think we should consider the need of BE work on export functionality.

Thanks for noticing this two problems. They are not directly related to changes in the PR so I will put a comment under my task for QA team to check those and create new bug tickets for them.

**Review by viktar-dzmitryieu-tao — APPROVED** [review]
Approved as this code works well and considering that the spotted issues will be addressed in some way

**Pull request merged**